compiler-cli/src/fs.rs

```diff
@@ -17,15 +17,23 @@ use std::{
     fmt::Debug,
     fs::File,
     io::{self, BufRead, BufReader, Write},
-    path::{Path, PathBuf},
     time::SystemTime,
 };
 
+use camino::{ReadDirUtf8, Utf8Path, Utf8PathBuf};
+
 use crate::{dependencies::UseManifest, lsp::LspLocker};
 
 #[cfg(test)]
 mod tests;
 
+/// This is a helper function for converting from `PathBuf` to `Utf8PathBuf`.
+/// We simply panic on non-utf8 paths.
+///
+pub fn utf8_or_panic(input: std::path::PathBuf) -> Utf8PathBuf {
+    Utf8PathBuf::from_path_buf(input).expect("Non Utf8 Path")
+}
```

Comment:
Sorry, I've been unclear. It was IO that should be moved into `compiler-cli`. This is a pure operation so there's no need to move it out of core, and I don't want to have a helper function that does nothing but call `expect`. We want to discourage `expect` so we don't have helpers that do it.